video/out/opengl/user_shaders.h

```diff
@@ -68,10 +73,32 @@ struct gl_user_shader {
     int components;
 };
 
-// Parse the next shader pass from `body`. The `struct bstr` is modified by the
+struct gl_user_shader_tex {
+    struct bstr name;
+    int w, h, d;
+    int components;
+    int bytes;
+    int mpgl_type;
+    GLenum gl_target;
+    GLenum gl_filter;
+    GLenum gl_border;
+    void *texdata;
+    // for video.c
+    GLenum gl_tex;
+};
+
+struct gl_user_shader {
+    enum user_shader_type type;
+    union {
+        struct gl_user_shader_hook hook;
+        struct gl_user_shader_tex tex;
+    } v;
```

Comment:
Tagged unions are a concept that doesn't exist in C. If you want to emulate it in C, at least don't do it in a "drill a hole into your skull" way. Which means you'd use a struct instead of an union to make it slightly more debugable. unions in C are for idiots who want to save a few bytes where it doesn't matter (like mpv_node).